Clarify that message id is not guaranteed to be unique from a processing perspective (#7127)
Particular/docs.particular.net: nservicebus/messaging/message-identity.md

---

> Each message dispatched from an [Endpoint](/nservicebus/endpoints/) has a unique identity. NServiceBus generates and stores this value in the `NServiceBus.MessageId` [header](/nservicebus/messaging/headers.md).

Reviewer: ```suggestion
Each message dispatched from an [Endpoint](/nservicebus/endpoints/) has a unique identity. NServiceBus generates and stores this value in the [`NServiceBus.MessageId` header](/nservicebus/messaging/headers.md#messaging-interaction-headers-nservicebus-messageid).
```

Many features take advantage of message identity. For example, the [Outbox](/nservicebus/outbox) uses message identity to deduplicate messages and [recoverability](/nservicebus/recoverability/) uses message identity to keep track of how many times the endpoint has tried to process a message.

> > [!INFO]

Author: ```suggestion
> [! NOTE]
```

> > [!WARNING]

Reviewer 2: What actionable thing does the user derive from this warning and why is it a warning and not an information level thing? I would rephrase it to something like

Author: I like the proposal

> > [!WARNING]
> > Message identities are not guaranteed to be unique from a processing perspective. For example, a published event's message identity will always be the same for all subscribers. To ensure a globally unique identity for processed messages, the endpoint name of the processing endpoint must be combined with the message identity.

Reviewer 2: ```suggestion
> [!INFO]
> Message identities are not guaranteed to be unique across endpoints from a processing perspective. For example, a published event’s message identity will always be the same for all subscribers. In cases where a globally unique identity for processed messages are required, the name of the processing endpoint should be combined with the message identity.
```
